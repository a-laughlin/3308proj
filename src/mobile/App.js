import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import {Svg} from 'expo';
const {Polyline } = Svg;
const data = [10,20,30,40,30,10,80,70]

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});


export default class App extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Text>Open up App.js to start working on your app!</Text>
        <Svg height="90%" width="100%" >
          <Polyline
              points={data.reduce((prev,y,i)=>prev+`${i*10},${y} `,"")}
              fill="none"
              stroke="black"
              strokeWidth="3"
          />
        </Svg>
      </View>
    );
  }
}
