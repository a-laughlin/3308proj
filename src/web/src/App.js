
/* eslint-disable no-unused-vars */
import React,{useState} from 'react';
import './App.css';
import {useHeartRateQuery,query,mutate,isLoading,isError,isData} from './state';
import {Div,Pre,Button,Polyline,Svg,Text,style,children,prop,state,onClick,toDstr,fromDstr
} from './hooks'
import {styleStringToObj as s} from './style-string-to-obj'
import data from './sample_data/ml_output_foo.json';
import {plog,pipe,get,oo,has,cond,is,ma,flatMap,pget} from './utils';
const text_style = {fill:'#000',transform:'translateY(20%)'}
const x_vals = [100,20,40,60,20,90,80,60,20,100,40,20,20,40,40]
const y_vals = [0,10,15,20,30,40,60,80,120,160,240,320,400,480]
const xy_vals = y_vals.map((y, x) => [x*10, 500-y].join(" "))


const beatsToXYStr = b=>b.map((y,x)=>`${x*10} ${100-y}`).join(' ');

export const SvgMain = (props)=>{
  const queryResult = useHeartRateQuery(props); // graphql query
  if(isLoading(queryResult)) {
    return <svg style={s`w100 h100 dB op.7`}><text>Loading...</text></svg>
  };
  if(isError(queryResult)) {
    return <svg style={s`w100 h100 dB op.7`}><text>`${queryResult}`</text></svg>
  };
  const [historical_hr,predicted_hr] = queryResult.hr; // destructure array
  return (
    <svg style={s`w100 h100 dB op.7`}>
      <rect style={s`w100 h100 fillF`}/>
      <polyline points={beatsToXYStr(predicted_hr.beats)} {...s`strk990 strkw1 fillT`}/>
      <polyline points={xy_vals} {...s`strkw1 strk009 fillT`}/>
      {x_vals.map((x,i) =>
        <text key={i} style={s(`fill0 transy${i*5}`)}>{x}</text>
      )}
      <text style={s`fill0 transy100`}>{JSON.stringify(data)}</text>
    </svg>);
}


const Svg2 = Svg(
  children(pipe(useHeartRateQuery,cond(
    [isLoading,Text('Loading...')],
    [isError,e=>Text(`${e}`)],
    [isData,({hr:[ {beats:hist,freq}, {beats:pred} ]}) =>[
      Text(freq,style(`fill009 transy10`)),
      Polyline(style(`strkw1 fillT strk009 transy10`), prop('points',p=>beatsToXYStr(hist))),
      Polyline(style(`strkw1 fillT strk990 transy50`), prop('points',p=>beatsToXYStr(pred))),
    ]]
  )))
);


const AdamExperiment2 = Div(
  state('rabbits','more',2),
  children(
    Svg2,
    p=>Div(`Rabbits:${p.rabbits}`,style('bg4BB fh fJCC')),
    ({more,rabbits})=>Button('More Rabbits!', onClick((btnProps,evt)=>more(rabbits*2))),
    p=>Button('More HeartRates!', onClick((_,evt)=>{
      // this updates the whole state... well, maybe only the part with hr
      // how update a field?
      // how update a list?
      mutate({query:'{hr {beats, freq, id, start, end }}'})(data=>({
        hr:[data.hr[0],{...data.hr[1],beats:[...data.hr[1].beats,50]}]
      }))
    })),
  ),
  style('w100 h100 bg299 t80 lh140 fv fAIC fJCSE usN crD'),
  oo('more')
)


export const App = props=>
// <img src="https://news.bitcoin.com/wp-content/uploads/2017/08/Markcap.png" alt="" style={s`w100 posA dB`} />
  <div {...props} style={s`taC`}>
    <div style={s`posF w100 h50 oH bgFFF`}><SvgMain/></div>
    <div style={s`posF w100 h50 oH bottom0 bg077`}><AdamExperiment2/></div>
  </div>;

export default App;
/* eslint-enable no-unused-vars */
