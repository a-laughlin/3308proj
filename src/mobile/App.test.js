import React from 'react';
import renderer from 'react-test-renderer';
import App from './App'

it('renders App', () => {
  const renderedApp = renderer.create(<App/>).toJSON();
  expect(renderedApp).toMatchSnapshot();
});
