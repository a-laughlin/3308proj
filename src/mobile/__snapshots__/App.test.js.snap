// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Open up App.js to start working on your app!
  </Text>
  <RNSVGSvgView
    bbHeight="90%"
    bbWidth="100%"
    height="90%"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        undefined,
        false,
        Object {
          "flex": 0,
          "height": "90%",
          "width": "100%",
        },
      ]
    }
    width="100%"
  >
    <RNSVGGroup
      fill={
        Array [
          0,
          0,
          0,
          0,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      font={Object {}}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={Array []}
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth="1"
      x={0}
      y={0}
    >
      <RNSVGPath
        d="M0 100 10 20 20 40 30 60 40 20 50 90 60 80 70 60 80 20 "
        fill={null}
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "stroke",
            "strokeWidth",
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={
          Array [
            0,
            0,
            0,
            0,
            1,
          ]
        }
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth="3"
        x={0}
        y={0}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
